packages/@aws-cdk/aws-ecs/lib/firelens-log-router.ts

```diff
@@ -203,32 +212,45 @@ export class FirelensLogRouter extends ContainerDefinition {
     if (options) {
       const enableECSLogMetadata = options.enableECSLogMetadata || options.enableECSLogMetadata === undefined;
       const configFileType = (options.configFileType === undefined || options.configFileType === FirelensConfigFileType.S3) &&
-        (cdk.Token.isUnresolved(options.configFileValue) || /arn:aws[a-zA-Z-]*:s3:::.+/.test(options.configFileValue))
+        (cdk.Token.isUnresolved(options.configFileValue) || /arn:aws[a-zA-Z-]*:s3:::.+/.test(options.configFileValue || ''))
         ? FirelensConfigFileType.S3 : FirelensConfigFileType.FILE;
-      this.firelensConfig = {
-        type: props.firelensConfig.type,
-        options: {
-          enableECSLogMetadata,
-          configFileType,
-          configFileValue: options.configFileValue,
-        },
-      };
-
-      // grant s3 access permissions
-      if (configFileType === FirelensConfigFileType.S3) {
-        props.taskDefinition.addToExecutionRolePolicy(new iam.PolicyStatement({
-          actions: [
-            's3:GetObject',
-          ],
-          resources: [options.configFileValue],
-        }));
-        props.taskDefinition.addToExecutionRolePolicy(new iam.PolicyStatement({
-          actions: [
-            's3:GetBucketLocation',
-          ],
-          resources: [options.configFileValue.split('/')[0]],
-        }));
+
+      if (options.configFileValue != undefined) {
+        this.firelensConfig = {
+          type: props.firelensConfig.type,
+          options: {
+            enableECSLogMetadata,
+            configFileType,
+            configFileValue: options.configFileValue,
+          },
+        };
+      } else {
+        this.firelensConfig = {
+          type: props.firelensConfig.type,
+          options: {
+            enableECSLogMetadata: enableECSLogMetadata,
+          },
+        };
       }
```

Comment:
```suggestion
      this.firelensConfig = {
        type: props.firelensConfig.type,
        options: {
        enableECSLogMetadata,
        ...(options.configFileValue ? {
          configFileType,
          configFileValue: options.configFileValue,
          } : {},
        ),
      }
```

Reply:
my editor strongly suggests this code is not valid